### ISSUE TITLE
[Feature] optional explicit frame size for inputs

### ISSUE THREAD

**Issue by beasteers** [bmcfee/pumpp#124]
#### Description
Pumps should be able to set a fixed frame size for inputs
```python 
pump.layers()
# size: (None, None, 256, 1)
pump.layers(frame_duration=8)
# size: (None, time_to_frames(8), 256, 1)
pump.layers(frame_length=32)
# size: (None, 32, 256, 1)
```

**Comment by bmcfee** [comment]
This seems like a bit of an anti-pattern to me.  Is the use-case here for fully connected models that don't support variable-length inputs?

My hesitation for supporting that would be that the transformers can't guarantee durations of inputs, and have no built-in mechanism for trimming.

Samplers *do* have such a mechanism though, so maybe it's ok?

**Comment by beasteers** [comment]
Yes it is. It came up in an autoencoder where I was getting `None` for a dense layer size, but maybe having temporally shifting features in a dense layer isn't a good idea anyways so I had a couple thoughts on how to get around it. 

The general usecase would be if we're using a sampler with a fixed size (`pump.sampler()`) and we know the size, we could support a larger family of models which require an explicit size. Up to you if you want to support it. Or we can just promote other workarounds 🤷‍♀️

**Comment by bmcfee** [comment]
Sorry to lag on this, but I do think it could be a useful feature in some form.  I'd like to think a bit more about the right API for it though, and probably collect a couple of example use-cases before committing to anything.
